app/routes.py

```diff
@@ -1,2 +1,184 @@
+
+@task_bp.route("", methods=["GET", "POST"])
+def handle_all_tasks():
+    tasks_response = []
+    
+    #WV2: Get Tasks Asc, Get Tasks Desc
+    if request.method == "GET":
+        if request.args.get('sort') == 'asc':
+            tasks = Task.query.order_by(Task.title.asc()).all()
+        elif request.args.get('sort') == 'desc':
+            tasks = Task.query.order_by(Task.title.desc()).all()
+
+        #WV1: No Saved Tasks, One Saved Tasks
+        else:
+            tasks = Task.query.all()     
+        for task in tasks:
+            tasks_response.append(task.to_dict())
+        return jsonify(tasks_response), 200 
+
+    #WV1: Create Task With None Completed_At, Create Task - Title, Create Task - Description, Create Task - Completed_At
+    #WV3: Create Task Wtih Valid Completed_At
+    if request.method == "POST":
+        request_body = request.get_json()
+        try:
+            new_task = Task(title = request_body["title"], 
+                            description = request_body["description"],
+                            completed_at = request_body["completed_at"])
+        except KeyError:
+            return ({'details': 'Invalid data'}, 400)        
+        db.session.add(new_task)
+        db.session.commit()
+        return jsonify({"task":new_task.to_dict()}), 201
+
+@task_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_single_task(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+
+    #WV1: Get Task, Get Task Not Found
+    if request.method == "GET":
+        if task is None:
+            return make_response("", 404)
+        return jsonify({"task": task.to_dict()}), 200 
+
+    #WV1: Update Task, Update Task Not Found
+    #WV3: Update Task With Valid Completed At
+    elif request.method == "PUT":
+        input_data = request.get_json()
+        if task is None:
+            return make_response("", 404)
+        task.title = input_data["title"]
+        task.description = input_data["description"]
+        db.session.commit()
+        return ({"task":task.to_dict()}), 200
+
+    #WV1: Delete Task, Delete Task Not Found
+    elif request.method == "DELETE":
+        if task is None:
+            return make_response("", 404)
+        else:
+            db.session.delete(task)
+            db.session.commit()
+            return ({'details': f'Task {task_id} "{task.title}" successfully deleted'}), 200
+
+#WV3: Mark Comp on Incom, Mark Incom on Comp, Mark Comp on Comp, Marck Comp on Missing, Mark Incom on Missing 
+@task_bp.route("/<task_id>/<completion_status>", methods=["PATCH"])
+def mark_complete(task_id, completion_status):
+
+    task = Task.query.get(task_id)
+    PATH="https://slack.com/api/chat.postMessage"
+    if task is None:
+        return make_response("", 404)
+    
+    if completion_status == "mark_complete":
+        task.completed_at = date.today()
+
+        #WV4: Send Slack Message
+        token=os.environ.get("TOKEN")   
```

Comment:
Nice job getting the token from the environment variables! You might consider fetching the variable from the OS once, and then keeping it saved so that your code does not need to do `os.environ.get` every time a task is completed. This could be achieved in a way similar to how the environment variables are handled in `app/__init__.py`. You might also consider a bit more descriptive of a environment variable name, perhaps something like `SLACK_API_TOKEN`.